src/Language/PureScript/Types.hs

```diff
@@ -590,7 +590,7 @@ compareType (TypeApp _ a b) (TypeApp _ a' b') = compareType a a' <> compareType
 compareType (TypeApp {}) _ = LT
 compareType _ (TypeApp {}) = GT
 
-compareType (ForAll _ a b c) (ForAll _ a' b' c') = compare a a' <> compareType b b' <> compare c c'
+compareType (ForAll _ a b c d) (ForAll _ a' b' c' d') = compare a a' <> fromMaybe LT (compareKind <$> b <*> b') <> compareType c c' <> compare d d'
```

Comment:
I'm not sure how I should factor in the possible kind annotation in to `eqType`/`compareType`.

Reply:
I don't think this is correct because that means if either side is Nothing then the answer is always `LT`. I would just write another function that does the pattern matching.

```haskell
compareMaybeKind :: Maybe (Kind a) -> Maybe (Kind a) -> Ordering
compareMaybeKind Nothing Nothing   = EQ
compareMaybeKind Nothing (Just _)  = LT
compareMaybeKind (Just _) Nothing  = GT
compareMaybeKind (Just a) (Just b) = compareKind a b
```

Reply:
Good point!